ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java

```diff
@@ -0,0 +1,212 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static week1.utils.TerminalUtils.calculateSumOfSoldProducts;
+import static week1.utils.TerminalUtils.findMaxPriceBill;
+import static week1.utils.TerminalUtils.findMinPriceBill;
+
+public class ITerminalControllerImpl implements ITerminalController {
+
+    private final static Logger logger = Logger.getLogger(ITerminalControllerImpl.class.getName());
+
+    private IAppDB iAppDB;
+
+    public ITerminalControllerImpl(IAppDB iAppDB) {
+        this.iAppDB = iAppDB;
+        logger.setLevel(Level.OFF);
+    }
+
+    @Override
+    public boolean login(String login, String password) {
+
+        iAppDB.setCurrentSeller(-1);
+
+        if ((login == null || password == null) ||
+                (login.equals("") || password.equals(""))
+                || iAppDB.getAllSellers().size() == 0) {
+
+            logger.warning("Login or pass is invalid or sellers is null!");
+            return false;
+        }
+
+        int sellerId = iAppDB.getAllSellers().
+                indexOf(iAppDB.findBySellerLoginAndPassword(login, password));
+
+        if (sellerId == -1) {
+            logger.warning("There aren't any sellers with such login\\pass!");
+            return false;
+        }
+
+        iAppDB.setCurrentSeller(sellerId);
+
+        logger.info("Successfully logged in!");
+        return true;
+    }
+
+    @Override
+    public Bill createBill() {
+
+        Bill bill = new Bill();
+
+        if (iAppDB.getCurrentSellerId() != -1)
+            bill.setSeller(iAppDB.getAllSellers().get(iAppDB.getCurrentSellerId()));
+
+        Bill billWithId = iAppDB.saveBill(bill);
+
+        logger.info("Method createBill done.");
+        return billWithId;
+    }
+
+    @Override
+    public Bill addProduct(int billId, Product product) {
+
+        Bill bill = iAppDB.findByBillId(billId);
+
+        if (bill == null) {
+            logger.warning("Not found a bill with such id");
+            return null;
+        }
+
+        product.setId(bill.getNextProductId());
+        bill.getProductList().add(product);
+        bill.setNextProductId(bill.getNextProductId() + 1);
+        bill.setAmountPrice(bill.getAmountPrice() + product.getPrice());
+
+        iAppDB.updateBill(bill);
+
+        logger.info("Method addProduct done");
+        return bill;
+    }
+
+    @Override
+    public List<Bill> getAllBills() {
+        return iAppDB.getAllBills();
+    }
+
+    @Override
+    public Bill closeBill(int billId) {
+
+        //TODO ADD SAAAAAAVE OF BILL
+        Bill bill = iAppDB.findByBillId(billId);
+
+        if (bill == null) {
+            logger.warning("Not found a bill with such id");
+            return null;
+        }
+
+        bill.setClosed(true);
+        bill.setCloseTime(LocalDateTime.now());
+
+        iAppDB.updateBill(bill);
+
+        logger.info("Bill was closed!");
+        return bill;
+    }
+
+    @Override
+    public Bill findBillById(int billId) {
+
```

Comment:
add validation

Reply:
What kind of validation? if id < 0 or what?

Reply:
https://github.com/presly808/aco22/pull/51/commits/b45056f67561831e9fe77997114005ebcd0e7a40 Fixed